usaspending_api/references/v2/views/city.py

```diff
@@ -0,0 +1,127 @@
+from django.conf import settings
+from rest_framework.response import Response
+from collections import OrderedDict
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.views import APIDocumentationView
+
+from usaspending_api.common.elasticsearch.client import es_client_query
+from usaspending_api.search.v2.elasticsearch_helper import es_sanitize
+from usaspending_api.common.validator.tinyshield import validate_post_request
+
+
+models = [
+    {
+        "name": "filter|country_code",
+        "key": "filter|country_code",
+        "type": "text",
+        "text_type": "search",
+        "optional": False,
+    },
+    {
+        "key": "filter|state_code",
+        "name": "fitler|state_code",
+        "type": "text",
+        "text_type": "search",
+        "optional": True,
+        "default": None,
+        "allow_nulls": True,
+    },
+    {
+        "key": "filter|scope",
+        "name": "filter|scope",
+        "type": "enum",
+        "enum_values": ("recipient_location", "primary_place_of_performance"),
+        "optional": False,
+    },
+    {"key": "search_text", "name": "search_text", "type": "text", "text_type": "search", "optional": False},
+    {"key": "limit", "name": "limit", "type": "integer", "max": 500, "optional": True, "default": 10},
+]
+
+
+@validate_post_request(models)
+class CityAutocompleteViewSet(APIDocumentationView):
+    """
+    endpoint_doc: usaspending_api/api_contracts/autocomplete/City.md
+    """
+
+    @cache_response()
+    def post(self, request, format=None):
+
+        search_text = es_sanitize(request.data["search_text"])
```

Comment:
Good idea to upper these input strings:
- search_text
- country_code
- state_code